src/containers/Tenant/QueryEditor/QueryEditorControls/OldQueryEditorControls.tsx

```diff
@@ -21,12 +28,19 @@ export const OldQueryEditorControls = ({
     onUpdateQueryMode,
     queryMode,
 }: QueryEditorControlsProps) => {
+    // On setting change when 'data' or 'query' option selected
+    useEffect(() => {
```

Comment:
I don't think this logic should be here. Component should not be aware about "another" modes. As for me, `queryMode` should be calculated in selector.